Use the Common Evidence Model #302

anthonysena opened the pull request:
- Updated EvidenceService.java to use the Common Evidence Model. Deprecated some methods that no longer apply.
- Refactored functionality for negative controls into their own namespace: org.ohdsi.webapi.evidence.negativecontrols
- Updated comments for javadocs/miredot

chrisknoll commented on pom.xml:
```diff
@@ -212,6 +212,30 @@
         <configuration>
            <licence>cHJvamVjdHxvcmcub2hkc2kuV2ViQVBJfDIwMjAtMDMtMzF8dHJ1ZXwtMSNNQzBDRkNNQ2R2ZEhjbDg0Z2tONVI3bHlmQ2Y4SnNkYUFoVUFoMkdtOUVFdy9RelpLVzM1V2p6a2FsdzJZV009</licence>
+           <restModel>
+						<httpStatusCodes>
```
Are these additions to change the way the service reports internal errors?

anthonysena replied:
Nice catch! These additions are to globally configure miredot to automatically document the response of an endpoint based on the return type used per this article:

http://www.miredot.com/docs/manual/rest-model-configuration/status-codes/

By configuring this in the pom.xml, we eliminate the following error at compile time:

```
Miredot is unable to document the status codes of rest interfaces, unless a statusCodes-block is specified in pom.xml
```

Also by configuring this globally, it eliminates the need to document this for each end-point. I bundled this in with this PR since I spent some energy documenting the Evidence service end-points. Let me know if you have any questions or concerns.

chrisknoll replied:
No, fair enough, just was curious. Thank you!

anthonysena commented:
Adding a patch for security to this branch; it is SQL Server specific and it encapsulated in https://github.com/OHDSI/WebAPI/pull/345/commits/1731170ca61dc55e2e4b099b810ce8530516e638

chrisknoll commented on src/main/resources/db/migration/oracle/V2.3.0.20180302143300__negative_control_redo.sql:
```diff
@@ -0,0 +1,45 @@
+    faers INTEGER,
+    user_excluded INTEGER,
+    user_included INTEGER,
+    CONSTRAINT PK_CONCEPT_SET_NEGATIVE_CONTROLS PRIMARY KEY (id)
```
This could be too long for oracle.

chrisknoll commented on src/main/resources/db/migration/oracle/V2.3.0.20180302143300__negative_control_redo.sql:
```diff
@@ -0,0 +1,45 @@
+    END IF;
+END;
+
+CREATE SEQUENCE ${ohdsiSchema}.CONCEPT_SET_NEGATIVE_CONTROLS_SEQUENCE MAXVALUE 9223372036854775807 NO CYCLE;
```
Check about assignign the start value of 1. I've seen some sequences start at 0...can you confirm (or just specify) that the sequencer will start at 1?  Some of our webapi calls use 0 to indicate 'new' status of record.

anthonysena commented:
I've moved the patch for security mentioned in this thread to a separate branch and opened as a PR (#387) since it was off-topic for these changes. I've addressed the notes regarding the Oracle sequence name length and initialization. We are still testing but I have re-based this branch off master and force pushed the update so if you have a branch locally, please remove it.
